crates/astria-sequencer/src/transaction/unsigned.rs

```diff
@@ -53,21 +84,76 @@ impl Transaction {
     }
 }
 
+#[async_trait::async_trait]
+impl ActionHandler for Transaction {
+    fn check_stateless(&self) -> Result<()> {
+        for action in &self.actions {
+            match action {
+                Action::AccountsAction(_) | Action::SecondaryAction(_) => {}
+            }
+        }
+        Ok(())
+    }
+
+    async fn check_stateful<S: StateReadExt + 'static>(
+        &self,
+        state: &S,
+        from: &Address,
+    ) -> Result<()> {
+        let curr_nonce = state.get_account_nonce(from).await?;
```

Comment:
Should this be also checking the stateful properties on the actions?

If so is there some sort of pattern we should be using to enforce this? I feel like if each added action needs a new line in stateless, stateful, and execute someone might later forget to do all of that and it should fail something.

Reply:
yeah, I omitted it + the `check_stateless` above because the checks don't actually do anything right now. but added because it should be there.

do you mean some sort of compile-time check?

Reply:
I think this is where something like the manager setup in cosmos is "correct" but we can refactor later, with only two things going on it's not necessary but as our action types grow it makes sense to have a system for ensuring each action is called.